docs/PLUGIN_AUTHORING.md

```diff
@@ -17,152 +17,144 @@ See the License for the specific language governing permissions and
 ```
-### Writing a publisher plugin
-A Snap publisher plugin allows publishing processed telemetry data into a variety of systems, databases, and monitors through Snap metrics. To compliant with metric types and plugin interfaces defined in Snap, a publisher plugin must implement the following methods:
+/acme/water/usage
+/acme/water/NewYork/usage
+/acme/water/NewYork/rainfall
+/acme/water/NewYork/Bronx/usage
+/acme/water/NewYork/Chelsea/usage
+/acme/water/Portland/usage
+/acme/water/Portland/rainfall
+/acme/water/Portland/Hollywood/usage
+/acme/water/Portland/Pearl/usage
 ```
-GetConfigPolicy() (*cpolicy.ConfigPolicy, error)
-Publish(contentType string, content []byte, config map[string]ctypes.ConfigValue) error
-```
-### Exposing a plugin
-Creating the main program to serve the newly written plugin as an external process in main.go. By defining "Plugin.PluginMeta" with plugin specific settings, the newly created plugin may have its setting to override Snap global settings. Please refer to [a sample](https://github.com/intelsdi-x/snap/blob/master/plugin/collector/snap-plugin-collector-mock1/main.go) to see how main.go is written. You may browse [snap global settings](https://github.com/intelsdi-x/snap/blob/master/snapd.go#L45-L119).
 
-Building main.go generates a binary executable. You may choose to sign the executable with our [plugin signing](https://github.com/intelsdi-x/snap/blob/master/docs/PLUGIN_SIGNING.md).
+A plugin can have any number of internal plugin namespace. It's up to the plugin author on how to organize the metrics in a meaningful way. This information should be documented in the README for ease of usage.
 
-### Localization
-All comments and READMEs within the plugin code should be in English.  For different languages, include appropriate translation files within the plugin package for internationalization.
+The metric namespace cannot contain [Unicode character U+1F422](http://www.fileformat.info/info/unicode/char/1f422/index.htm) i.e. :turtle:
 
-### README
-All plugins should have a README with some standard fields:
-```
- 1. Snap version requires at least
- 2. Snap version tested up to
- 3. Supported platforms
- 4. Contributor
- 5. License
-```
-### Encryption
-Snap provides the encryption capability for both HTTP and TCP clients. The communication between the Snap daemon and the plugins is encrypted by default. Should you want to disable the encrypted communication, when authoring a plugin, use the `Unsecure` option for your plugin's meta:
-```
-//Meta returns the metadata for MyPlugin
-func Meta() *plugin.PluginMeta {
-    return plugin.NewPluginMeta(name, ver, type, ct, ct2, plugin.Unsecure(true))
+NOTE: The intelsdi-x project reserves the org namespace `/intel`.
+
+### Plugin Interface
+
+Depending on the type of plugin, they must implement several methods to satisfy the appropriate interfaces. Please see the plugin library for language specific examples and documentation.
```

Comment:
Is it possible to create a title under "overview" called "plugin libraries", and you can link to this title from this sentence. Or is it too much?